compiler/util-io/src/org/jetbrains/kotlin/konan/file/ZipUtil.kt

```diff
@@ -46,7 +53,13 @@ fun <T> File.withZipFileSystem(mutable: Boolean = false, action: (FileSystem) ->
     return try {
         action(zipFileSystem)
     } finally {
-        zipFileSystem.close()
+        val counter = counters[zipFileSystem]!!
+        if (counter == 0) {
```

Comment:
Why 0? We're decrementing after this check, so probably here we need to compare with 1.

Reply:
Uh, seems to be a stale cherry-pick. We've already fixed it all somewhere. Let me force push.

Reply:
There you go.